### PR TITLE
(SUP-2848) st0361a_uploading_facts convert to ruby

### PR DESCRIPTION
**Pull request by MartyEwings**
Removes bash and power shell versions
adds spec tests

### CODE REVIEW

**Comment by MartyEwings** [comment]
@m0dular are you able to approve? ideally we wouldn't "Shell out" but given this is run infrequently, and the general improvements we have since moving from BASH and powershell this is a positive step

File: tasks/st0361_uploading_facts.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+puppet_bin_dir = if Facter.value(:osfamily).eql? 'windows'
+                   'C:\\"Program Files"\\"Puppet Labs"\\Puppet\\bin\\'
+                 else
+                   '/opt/puppetlabs/bin/'
+                 end
+
+factsupload = `#{puppet_bin_dir}puppet facts upload`
+
+puts factsupload
```
Comment (jarretlavallee) [inline tasks/st0361_uploading_facts.rb:13]: This should be structured output based on the exit status of the command.

Comment (MartyEwings) [reply tasks/st0361_uploading_facts.rb:13]: Yeah this is what i did after i spent an hour failing at doing a proper structured return :) but start as i mean to go on! ill give it another shot

Comment (MartyEwings) [reply tasks/st0361_uploading_facts.rb]: OK should be complete

**Review by jarretlavallee — APPROVED** [review]
👍

**Pull request merged**